exla/lib/exla/mlir/module.ex

```diff
@@ -11,11 +11,22 @@ defmodule EXLA.MLIR.Module do
   alias EXLA.Executable
   alias EXLA.Shape
 
+  @context_key {__MODULE__, :mlir_context}
+  def init do
+    :persistent_term.put(@context_key, EXLA.NIF.new_mlir_context() |> unwrap!())
+  end
+
   @doc """
   Creates a new MLIR module.
   """
   def new() do
-    ref = EXLA.NIF.new_mlir_module() |> unwrap!()
+    context = :persistent_term.get(@context_key, nil)
+
+    unless context do
+      raise "No MLIR context found"
+    end
```

Comment:
We could do the init lazily:

```elixir
defp get_or_create_context() do
  if context = :persistent_term.get(@context_key, nil) do
    context
  else
    context = EXLA.NIF.new_mlir_context() |> unwrap!()
    :persistent_term.put(@context_key, context)
    context
  end
end
```

In the very unlikely race condition, where `new/0` is called concurrently we may create two contexts, which should be fine.

@josevalim?

Reply:
I changed exactly because there was a race condition

Reply:
Ok, nevermind then :D